analyzers/src/SonarAnalyzer.CSharp/Rules/TooManyLoggingCalls.cs

```diff
@@ -0,0 +1,217 @@
+    {
+        public const int Debug = 4;
+        public const int Error = 1;
+        public const int Information = 2;
+        public const int Warning = 1;
+    }
+
+    public static readonly DiagnosticDescriptor Rule = DescriptorFactory.Create(DiagnosticId, MessageFormat, isEnabledByDefault: false);
+
+    private static readonly ImmutableArray<KnownType> LoggerTypes = ImmutableArray.Create(
+        KnownType.Castle_Core_Logging_ILogger,
+        KnownType.log4net_ILog,
+        KnownType.log4net_Util_ILogExtensions,
+        KnownType.Microsoft_Extensions_Logging_ILogger,
+        KnownType.Microsoft_Extensions_Logging_LoggerExtensions,
+        KnownType.NLog_ILogger,
+        KnownType.NLog_ILoggerBase,
+        KnownType.NLog_ILoggerExtensions,
+        KnownType.Serilog_ILogger,
+        KnownType.Serilog_Log);
+
+    private static readonly ImmutableArray<LoggingCategory> LoggingCategories = ImmutableArray.Create<LoggingCategory>(
+        new(CategoryNames.Debug, ImmutableHashSet.Create("ConditionalDebug", "ConditionalTrace", "Debug", "DebugFormat", "LogDebug", "LogTrace", "Trace", "TraceFormat", "Verbose")),
+        new(CategoryNames.Error, ImmutableHashSet.Create("Error", "ErrorFormat", "Fatal", "FatalFormat", "LogCritical", "LogError")),
+        new(CategoryNames.Information, ImmutableHashSet.Create("Info", "InfoFormat", "Information", "LogInformation")),
+        new(CategoryNames.Warning, ImmutableHashSet.Create("LogWarning", "Warn", "WarnFormat", "Warning")));
+
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+    [RuleParameter("debugThreshold", PropertyType.Integer, "The maximum number of DEBUG, TRACE and VERBOSE statements allowed in the same code block.", DefaultTresholds.Debug)]
+    public int DebugThreshold { get; set; } = DefaultTresholds.Debug;
+
+    [RuleParameter("errorThreshold", PropertyType.Integer, "The maximum number of ERROR and FATAL statements allowed in the same code block.", DefaultTresholds.Error)]
+    public int ErrorThreshold { get; set; } = DefaultTresholds.Error;
+
+    [RuleParameter("informationThreshold", PropertyType.Integer, "The maximum number of INFORMATION statements allowed in the same code block.", DefaultTresholds.Information)]
+    public int InformationThreshold { get; set; } = DefaultTresholds.Information;
+
+    [RuleParameter("warningThreshold", PropertyType.Integer, "The maximum number of WARNING statements allowed in the same code block.", DefaultTresholds.Warning)]
+    public int WarningThreshold { get; set; } = DefaultTresholds.Warning;
+
+    protected override void Initialize(SonarParametrizedAnalysisContext context) =>
+        context.RegisterCompilationStartAction(cc =>
+        {
+            if (cc.Compilation.ReferencesAny(KnownAssembly.CastleCore, KnownAssembly.Log4Net, KnownAssembly.MicrosoftExtensionsLoggingAbstractions, KnownAssembly.Serilog, KnownAssembly.NLog))
```

Comment:
Nitpick:
I would put these in popularity order as well,
Microsoft -> NLog -> Serilog -> Log4net -> CastleCore

Reply:
Nitpick again, optional:
You could extract the `KnownAssemblies` to an array, just for readability. Up to you.